lib/std/builtin.zig

```diff
@@ -664,16 +664,17 @@ pub const PanicFn = fn ([]const u8, ?*StackTrace) noreturn;
 
 /// This function is used by the Zig language code generation and
 /// therefore must be kept in sync with the compiler implementation.
-pub const panic: PanicFn = if (@hasDecl(root, "panic")) root.panic else default_panic;
+pub const panic: PanicFn = if (@hasDecl(root, "panic"))
+    root.panic
+else if (@hasDecl(root, "os") and @hasDecl(root.os, "panic"))
+    root.os.panic
```

Comment:
This case has no cold cache hint in contrast to the original code.

Reply:
I don't understand. `@setCold(true)` is still the first line in `default_panic`.

Perhaps you aren't recognizing that this if statement is comptime? Or maybe I am totally missing your point?

Reply:
my bad, you are correct here.